rmf_site_editor/src/site/scenario.rs

```diff
@@ -102,65 +135,214 @@ pub fn update_current_scenario(
+
+                match update.update_type {
+                    UpdateInstanceType::Include => {
+                        let instance_pose = instance.pose().or(recall_instance
+                            .get(*scenario_child)
+                            .ok()
+                            .and_then(|r| r.pose));
 
-                let existing_moved_instance = current_scenario
-                    .moved_instances
-                    .iter_mut()
-                    .find(|(e, _)| *e == entity);
-                if let Some(existing_moved_instance) = existing_moved_instance {
-                    existing_moved_instance.1 = pose.clone();
-                    return;
-                } else {
-                    current_scenario
-                        .moved_instances
-                        .push((entity, pose.clone()));
-                    return;
+                        if parent_pose.is_some() {
+                            *instance = Instance::new_inherited(instance_pose)
+                        } else if let Some(instance_pose) = instance_pose {
+                            *instance = Instance::new_added(instance_pose);
+                        } else {
+                            let instance_id = model_instances
+                                .get(update.instance)
+                                .map(|n| n.0.clone())
+                                .unwrap_or(format!("{}", update.instance.index()).to_string());
+                            error!(
+                                "Unable to retrieve pose for instance {:?}, \
+                                setting to default pose as AddedInstance in current scenario",
+                                instance_id
+                            );
+                            *instance = Instance::new_added(Pose::default());
+                        }
+                    }
+                    UpdateInstanceType::Hide => {
+                        *instance = Instance::new_hidden();
+                    }
+                    UpdateInstanceType::Modify(new_pose) => {
+                        // Update Pose changes
+                        match instance {
+                            Instance::Added(_) => *instance = Instance::new_added(new_pose.clone()),
+                            Instance::Inherited(_) => {
+                                *instance = Instance::new_inherited(Some(new_pose.clone()))
+                            }
+                            _ => {}
+                        }
+                    }
+                    UpdateInstanceType::ResetPose => {
+                        if parent_pose.is_some() {
+                            *instance = Instance::new_inherited(None)
+                        }
+                    }
+                    _ => {}
                 }
             }
         }
+        if current_scenario.0.is_some_and(|e| e == update.scenario) {
+            change_current_scenario.send(ChangeCurrentScenario(update.scenario));
```

Comment:
It's not totally clear to me if this is necessary. I could imagine a plugin that wants to make modifications to scenarios without changing what the current scenario is.

Reply:
The `ChangeCurrentScenario` event reader updates the `CurrentScenario` resource to the contained scenario entity [here](https://github.com/open-rmf/rmf_site/blob/xiyu/include_instances/rmf_site_editor/src/site/scenario.rs#L121). This filter prevents unwanted scenario switching.

Reply:
Ohhh I see now. Then I guess the purpose of the `ChangeCurrentScenario` event is to force an update of the visuals rather than actually changing the current scenario?

Reply:
That's right, it can be interpreted as changing elements in the current scenario